alchemy/src/cloudflare/secret.ts

```diff
@@ -0,0 +1,218 @@
+import type { Context } from "../context.ts";
+import { Resource, ResourceKind } from "../resource.ts";
+import { secret as alchemySecret, type Secret as AlchemySecret } from "../secret.ts";
+import { handleApiError } from "./api-error.ts";
+import {
+  createCloudflareApi,
+  type CloudflareApi,
+  type CloudflareApiOptions,
+} from "./api.ts";
+import type { SecretsStore } from "./secrets-store.ts";
+
+/**
+ * Properties for creating or updating a Secret in a Secrets Store
+ */
+export interface SecretProps extends CloudflareApiOptions {
+  /**
+   * The secrets store to add this secret to
+   */
+  store: SecretsStore<any>;
+
+  /**
+   * The secret value to store
+   * Can be a string or an existing Secret instance
+   * 
+   * @example
+   * ```ts
+   * const secret = await Secret("api-key", {
+   *   store: mySecretsStore,
+   *   value: alchemy.secret(process.env.API_KEY)
+   * });
+   * ```
+   */
+  value: string | AlchemySecret;
```

Comment:
It must be a secret and can't be a string for security reasons. 

If we want to support this flexible interface, you must not expose the Secret resource directly. 

Instead expose a function that will coalesce this into a Secret before it passes to the resource. 

Use _Secret for the non-exported resource. 

The reason is that any properties passed into a resource will be serialized. If it's not a secret by that point it will be stored in plain text. 

The mapping function solves this by providing an open ended interface and coalescing to secure.